Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs

```diff
@@ -146,30 +150,44 @@ void LocationAuthCallback(object sender, CLAuthorizationChangedEventArgs e)
                     if (e.Status == CLAuthorizationStatus.NotDetermined)
                         return;
 
-                    if (previousState == CLAuthorizationStatus.AuthorizedWhenInUse && !whenInUse)
+                    try
```

Comment:
This double exception swallow is a bit scary... What is this for? Can we pass it to the tcs?

Reply:
Yeah, I can do that especially if the tcs isn't null :)

Basically ... if you look at #1238 who knows that is happening so wanted to just catch it all!